pkg/channel/distributed/controller/kafkachannel/reconciler.go

```diff
@@ -260,21 +264,30 @@ func (r *Reconciler) configMapObserver(logger *zap.SugaredLogger, configMap *cor
 		return
 	}
 
-	// Merge The ConfigMap Settings Into The Provided Config
-	saramaSettingsYamlString := configMap.Data[testing.SaramaSettingsConfigKey]
+	// Get The Sarama Config Yaml From The ConfigMap
+	saramaSettingsYamlString := configMap.Data[configconstants.SaramaSettingsConfigKey]
+
+	// Create A Kafka Auth Config From Current Credentials (EnvVars From Secret Take Precedence Over ConfigMap)
+	var kafkaAuthCfg *client.KafkaAuthConfig
+	if r.kafkaUsername != "" {
+		kafkaAuthCfg = &client.KafkaAuthConfig{
+			SASL: &client.KafkaSaslConfig{
+				User:     r.kafkaUsername,
+				Password: r.kafkaPassword,
+			},
+		}
+	}
 
-	// Load the Sarama settings from our configmap, ignoring the eventing-kafka result.
+	// Build A New Sarama Config With Auth From Secret And YAML Config From ConfigMap
 	saramaConfig, err := client.NewConfigBuilder().
 		WithDefaults().
+		WithAuth(kafkaAuthCfg).
```

Comment:
I think that if the kafkaAuthCfg is left at its default (nil) value (because r.kafkaUsername == "" for a no-auth setup), then this builder will use whatever SASL (User/Password) settings are in the configmap, which could override the "no-auth" setup that's already present in the Reconciler struct (because the Build() function only changes the TLS/SALS settings if b.auth != nil, which is after the yaml is unmarshaled wholesale into the sarama.Config).

If we don't want the incoming configmap to be able to overwrite a no-auth setup, this might need some more logic.

Reply:
Yeah, I think the way we'd want to address this is to change the Builder's `Build()` function to have an `else` clause when there's no Auth specified to the Builder to clear any config that might have been in the YAML.  Something like...
```
	// then apply auth settings
	if b.auth != nil {
	...
	} else {
		// no auth specified so clear any auth from YAML (only accept explicit auth from builder)
		defaultConfig := sarama.NewConfig()
		config.Net.TLS = defaultConfig.Net.TLS
		config.Net.SASL = defaultConfig.Net.SASL
	}
```

It's really just a question of how we want the builder to operate.  We can change in a separate PR if desired.

Would love to hear what @aliok thinks about the issue as well.